.github/workflows/build-and-push.yml

```diff
@@ -38,15 +41,61 @@ jobs:
             password: ${{ secrets.GITHUB_TOKEN }}
 
         - name: Build and push
+          env:
+            REGISTRY_INPUT: ${{ inputs.push_to_project }}
+            GH_REF: ${{ inputs.ref }}
+          working-directory: src/github.com/cloudbase/garm
           run: |
-            cd src/github.com/cloudbase/garm
-            VERSION=$(git describe --tags --match='v[0-9]*' --always ${{ github.event.inputs.ref }})
+            get_gh_latest_release() {
+              curl -s -L -H "Accept: application/vnd.github+json" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              "https://api.github.com/repos/$1/latest" \
+              | jq -r '.tag_name'
+            }
+            set -x
+            if [ "$GH_REF" == "main" ]; then
+              VERSION="nightly"
+              AZURE_REF="main"
+              OPENSTACK_REF="main"
+              LXD_REF="main"
+              INCUS_REF="main"
+              AWS_REF="main"
+              GCP_REF="main"
+              EQUINIX_REF="main"
+              LINODE_REF="main"
+              K8S_REF="main"
+            else
+              VERSION=$(git describe --tags --match='v[0-9]*' --always)
+              AZURE_REF=$(get_gh_latest_release cloudbase/garm-provider-azure)
+              OPENSTACK_REF=$(get_gh_latest_release cloudbase/garm-provider-openstack)
+              LXD_REF=$(get_gh_latest_release cloudbase/garm-provider-lxd)
+              INCUS_REF=$(get_gh_latest_release cloudbase/garm-provider-incus)
+              AWS_REF=$(get_gh_latest_release cloudbase/garm-provider-aws)
+              GCP_REF=$(get_gh_latest_release cloudbase/garm-provider-gcp)
+              EQUINIX_REF=$(get_gh_latest_release cloudbase/garm-provider-equinix)
+              LINODE_REF=$(get_gh_latest_release flatcar/garm-provider-linode)
+              K8S_REF=$(get_gh_latest_release mercedes-benz/garm-provider-k8s)
+            fi
+            if [ "$GH_REF" == "release/v1" ]; then
```

Comment:
The release branches are called: `release/v0.1` and `release/v0.2`. You can see the names of the branches we merge into before release, in the "switch branches/tags" dropdown menu.